cpp/src/inflight_requests.cpp

```diff
@@ -47,7 +47,19 @@ void InflightRequests::remove(const Request* const request)
       return;
     } else if (result == -1) {
       auto search = _inflightRequests->find(request);
-      if (search != _inflightRequests->end()) _inflightRequests->erase(search);
+      decltype(search->second) tmpRequest;
```

Comment:
**question:** Would the need for this be obviated by `InflightRequests::remove` taking a `shared_ptr<Request>` to remove, rather than raw pointer? Or is that not possible because it's stored in a C data structure somewhere?

Reply:
Maybe, early on this was definitely a design choice because of potential circular dependencies, it may not be necessary anymore. Perhaps we should consider redesigning `InflightRequests` to use `std::set` instead of a `std::map`, it might be sufficient now. I'd prefer to keep this to a future time though, as it's more important to first get to test UCXX with the remaining of RAPIDS.

What do you think?

Reply:
Yes, that makes sense.